Vanjs examples

thednp opened the pull request:
Added VanJS examples.

@brillout I figured out and made a separate branch.

Notes:
- I forked the react example and adapted it for 2 VanJS examples: simple and JSX
- I tried to update the tests, you might want to look into that
- let me know if any issue.

Hope this works :pray:

brillout commented:
Neat! Le me know when this PR is ready and I'll have a look at this.

thednp commented:
@brillout I'll fix the tests and will bump you again.

How can I filter e2e tests to only include the new examples?

brillout commented:
Since we'll eventually be removing the `/examples/` directory it's best we move the VanJS examples outside the `vikejs/vike` repo. We can keep them in here until I review the examples (quite curious to have a look at them) but you can already remove the `/examples/van-*` tests.

thednp commented:
@brillout I've renamed the test files in `examples/van-*`, most of them should work, some need small updates. Since we put some work into them, I think it's best to keep them.

So I've renamed files from `.fileName.ts` => `._fileName.ts`.
If later you decide to remove them, it's your take.

thednp commented:
@brillout I\m unable to solve this last commit validation error:

```
[16:54:22.348][/docs/.test-preview.test.ts][pnpm run preview][stderr] Error: URL /van not found in following URLs:
```

Is this important?

thednp commented:
@brillout ready to play!

thednp commented:
@brillout all things done, you can merge.

brillout commented:
I love how simple VanJS is :heart_eyes:

It's been a long time I think there is a dire need for a simple JSX (or Vanilla JS) design with built-in reactivity. What's your marketing plan to popularize VanJS?

As for this PR, let's (re)move the `examples/van-*` (the issue is that Vike's CI is already too long for Vite's ecosystem CI) outside this repo and this can be merged then.

We can then think of next steps. For example, a `vike-van` could actually make sense if you ask me — it doesn't have to be maintained for now (although the best would be to eventually have a highly polished & maintained `vike-van`).

As for Bati, the issue is that it makes maintaining Bati harder. That's why, so far, I'm inclined to have a fork instead of adding Van to it but that can be discussed.

brillout commented:
> I love how simple VanJS is 😍
> 
> It's been a long time I think there is a dire need for a simple JSX (or Vanilla JS) design with built-in reactivity. What's your marketing plan to popularize VanJS?

Created a discussion about it: https://github.com/vanjs-org/van/discussions/427.

brillout commented:
Merged https://github.com/vikejs/vike/commit/81c05f6c9b2fec90038fcc667319092d1f9ffe43.
